src/Raven.Server/Commercial/SetupManager.cs

```diff
@@ -0,0 +1,365 @@
+                        onProgress(progress);
+                        return null;
+                    }
+
+                    var i = 0;
+                    while (true)
+                    {
+                        await Task.Delay(1000, token);
+                        response = await ApiHttpClient.Instance.PostAsync("/api/v4/dns-n-cert/registration-result",
+                                new StringContent(JsonConvert.SerializeObject(setupInfo), Encoding.UTF8, "application/json"), token)
+                            .ConfigureAwait(false);
+
+                        var registrationResult = JsonConvert.DeserializeObject<RegistrationResult>(await response.Content.ReadAsStringAsync().ConfigureAwait(false));
+                        
+                        if (registrationResult.Status == RegistrationStatus.Error)
+                        {
+                            var responseString = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                            progress.AddError("Stage2: Failed. Cannot complete setup. " + responseString);
+                            onProgress(progress);
+                            return null;
+                        }
+                        if (registrationResult.Status == RegistrationStatus.Pending)
+                        {
+                            switch (i)
+                            {
+                                case 30:
+                                    progress.AddInfo("Stage2: Still Waiting...");
+                                    break;
+                                case 60:
+                                    progress.AddInfo("Stage2: Still Waiting... hang in there, just a few more seconds.");
+                                    break;
+                                case 120:
+                                    progress.AddInfo("Stage2: Something is wrong. You may wait a little longer but you might as well abort and try again.");
+                                    break;
+                                default:
+                                    progress.AddInfo("Stage2: Waiting...");
+                                    break;
+                            }
+                            onProgress(progress);
+                        }
+                        else if (registrationResult.Status == RegistrationStatus.Done)
+                            break;
+                        i++;
+                    }
+
+                    progress.Processed++;
+                    progress.AddInfo("Stage2: Finished.");
+                    progress.AddInfo($"Stage3: Completing Let's Encrypt challenge for domain {setupInfo.Domain}.");
+                    onProgress(progress);
+                    
+                    var challengeResult = await acmeClient.CompleteChallenge(challenge);
+
+                    for (i = 0; i < 15; i++)
```

Comment:
Why limit to 15 retries? Have the same logic as with the api server.
And make sure that we log things and send progress report.